dotnet-desktop-guide/net/winforms/printing/how-to-print-text-document.md

```diff
@@ -0,0 +1,53 @@
+  - "Windows Forms, printing text"
+  - "printing [Windows Forms], text"
+ms.custom: devdivchpfy22
+---
+
+# How to print a multi-page text file (Windows Forms .NET)
+
+It's common for Windows-based applications to print text. The <xref:System.Drawing.Graphics> class provides methods for drawing objects (graphics or text) to a device, such as a screen or printer. The following section describes in detail the process to print text file. This method doesn't support printing non-plain text files, such as an Office Word document or a _PDF_ file.
+
+> [!NOTE]
+> The <xref:System.Windows.Forms.TextRenderer.DrawText%2A> methods of <xref:System.Windows.Forms.TextRenderer> are not supported for printing. You should always use the <xref:System.Drawing.Graphics.DrawString%2A> methods of <xref:System.Drawing.Graphics>, as shown in the following code example, to draw text for printing purposes.
+
+## To print text
+
+01. Add a <xref:System.Drawing.Printing.PrintDocument> component.
+
+01. Add a string variable to the class.
+
+    :::code language="csharp" source="snippets/how-to-print-text-document/csharp/Form1.cs" id="add_string_to_your_form":::
+
+    :::code language="vb" source="snippets/how-to-print-text-document/vb/Form1.vb" id="add_string_to_your_form":::
+
+01. To print a document, set the <xref:System.Drawing.Printing.PrintDocument.DocumentName%2A> property to the document you wish to print. Then open and read the document content to the string you added previously. Code to open and read document can be added to the button click event handler. Call the <xref:System.Drawing.Printing.PrintDocument.Print%2A> method to raise the <xref:System.Drawing.Printing.PrintDocument.PrintPage> event. As shown in the highlighted code below, the method `.Print()` will start printing and method raise the `PrintPage` event.
+
+    :::code language="csharp" source="snippets/how-to-print-text-document/csharp/Form1.cs" id="set_DocumentName_and_string" highlight= "10-12":::
+
+    :::code language="vb" source="snippets/how-to-print-text-document/vb/Form1.vb" id="set_DocumentName_and_string" highlight= "17-19":::
+
+01. Select the `PrintDocument` component in the Visual Designer. On the **Properties** pane, select the **Event** filter and then double-click the `PrintPage` event to generate an event handler.
+
+01. In the <xref:System.Drawing.Printing.PrintDocument.PrintPage> event handler, use the <xref:System.Drawing.Printing.PrintPageEventArgs.Graphics%2A> property of the <xref:System.Drawing.Printing.PrintPageEventArgs> class and the document contents to calculate line length and lines per page. After each page is drawn, check if it's the last page, and set the <xref:System.Drawing.Printing.PrintPageEventArgs.HasMorePages%2A> property of the `PrintPageEventArgs` accordingly. The `PrintPage` event is raised until `HasMorePages` is `false`. Also, make sure the `PrintPage` event is associated with its event-handling method.
```

Comment:
The "make sure the event is associated" doesn't need to be there anymore because the previous step added it.
```suggestion
01. In the <xref:System.Drawing.Printing.PrintDocument.PrintPage> event handler, use the <xref:System.Drawing.Printing.PrintPageEventArgs.Graphics%2A> property of the <xref:System.Drawing.Printing.PrintPageEventArgs> class and the document contents to calculate line length and lines per page. After each page is drawn, check if it's the last page, and set the <xref:System.Drawing.Printing.PrintPageEventArgs.HasMorePages%2A> property of the `PrintPageEventArgs` accordingly. The `PrintPage` event is raised until `HasMorePages` is `false`.
```